components/org.wso2.carbon.identity.conditional.auth.functions.session/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/session/util/SessionValidationUtil.java

```diff
@@ -0,0 +1,155 @@
+        requestData.put(SessionValidationConstants.START_TAG, SessionValidationConstants.START_INDEX);
+        requestData.put(SessionValidationConstants.COUNT_TAG,
+                SessionValidationConstants.SESSION_COUNT_MAX);
+        StringEntity entity = new StringEntity(requestData.toString(), ContentType.APPLICATION_JSON);
+        HttpPost httpRequest = new HttpPost(IdentityUtil.getProperty(SessionValidationConstants
+                .TABLE_SEARCH_CONFIG_NAMES));
+        String toEncode = IdentityUtil.getProperty(SessionValidationConstants.USERNAME_CONFIG_NAME)
+                + SessionValidationConstants.ATTRIBUTE_SEPARATOR
+                + IdentityUtil.getProperty(SessionValidationConstants.PASSWORD_CONFIG_NAME);
+        byte[] encoding = Base64.encodeBase64(toEncode.getBytes(Charset.forName(StandardCharsets.UTF_8.name())));
+        String authHeader = new String(encoding, Charset.defaultCharset());
+        //Adding headers to request
+        httpRequest.addHeader(HTTPConstants.HEADER_AUTHORIZATION, SessionValidationConstants.AUTH_TYPE_KEY +
+                authHeader);
+        httpRequest.addHeader(SessionValidationConstants.CONTENT_TYPE_TAG, ContentType.APPLICATION_JSON.toString());
+        httpRequest.setEntity(entity);
+        return httpRequest;
+    }
+
+    /**
+     * Method to get Query string from the user details.
+     *
+     * @param tenantDomain Tenant domain of the user
+     * @param username     username of the user
+     * @param userStore    user store of the user
+     * @return Query string with user information
+     */
+    public static String getQuery(String tenantDomain, String username, String userStore) {
+
+        return
+                SessionValidationConstants.TENANT_DOMAIN_TAG +
+                        SessionValidationConstants.ATTRIBUTE_SEPARATOR +
+                        tenantDomain +
+                        SessionValidationConstants.AND_TAG +
+                        SessionValidationConstants.USERNAME_TAG +
+                        SessionValidationConstants.ATTRIBUTE_SEPARATOR +
+                        username +
+                        SessionValidationConstants.AND_TAG +
+                        SessionValidationConstants.USER_STORE_TAG +
+                        SessionValidationConstants.ATTRIBUTE_SEPARATOR +
+                        userStore;
+    }
+
+    /**
+     * Method used for adding authentication header for httpMethod.
+     *
+     * @param httpMethod httpMethod that needs auth header to be added
+     * @param username   username of user
+     * @param password   password of the user
+     */
+    public static HttpPost setAuthorizationHeader(HttpPost httpMethod, String username, String password) {
+
+        String toEncode = username + SessionValidationConstants.ATTRIBUTE_SEPARATOR + password;
+        byte[] encoding = org.apache.commons.codec.binary.Base64.encodeBase64(toEncode.getBytes(Charset.forName(StandardCharsets.UTF_8.name())));
```

Comment:
do we need the FQN of 'Base64' class here? check and try to import

Reply:
imported in the new commit